GoogleForCreators/web-stories-wp
# Added trimmer component

## Context

This adds an actual trimmer component to the trim mode.

## To-do

_Copied from #8890._

- [x] Implement the proper design
- ~Fill the timeline with screenshots taken from the actual video at appropriate intervals (probably at a fixed % of the video - maybe 10 pictures evenly distributed?)~ ➡️   #9104
- [x] The two buttons only show up once the user has moved either handle
- [x] The cancel button resets the endpoints to the values they had when the trim mode was initiated (not to the ends necessarily, just back to where they were)
- [x] The trim button does nothing yet ➡️   #9105
- Maybe pause the video while dragging handles ➡️   #9108 

## User-facing changes

![trimmer-initial-2](https://user-images.githubusercontent.com/637548/134282556-c87bd616-4a25-4fad-af86-05eea1f49d66.gif)

For tight-fitting pages (either fit or on a zoom level), the trimming rail has a max width smaller than the actual page to make sure the is room for buttons on either side:

| No buttons | Buttons | Zoom |
|-|-|-|
| ![Screen Shot 2021-09-23 at 23 20 28](https://user-images.githubusercontent.com/637548/134613358-275b1a2e-634c-4245-aad9-4db4c02f8c34.png) | ![Screen Shot 2021-09-23 at 23 20 15](https://user-images.githubusercontent.com/637548/134613363-5ac9b53e-2271-490f-bea7-357d7f5772a9.png) | ![Screen Shot 2021-09-23 at 23 19 54](https://user-images.githubusercontent.com/637548/134613366-3b6a9931-e276-49a6-b5e0-48977cd3f68f.png) |

## Testing Instructions

This PR can be tested by following these steps:

1. Add a video (background or not) to a page
2. Enter trim mode (from quick-action or design panel)
3. Use the mouse to move the trimming handles
4. Use the keyboard to move the trimming handles
5. Observe that things work as expected
5. Press "Cancel" to abort trimming

## Checklist




---

Fixes #8890
